src/target.c

```diff
@@ -193,7 +194,7 @@ stumpless_add_entry( struct stumpless_target *target,
   VALIDATE_ARG_NOT_NULL_INT_RETURN( target );
   VALIDATE_ARG_NOT_NULL_INT_RETURN( entry );
 
-  if( unlikely( !target->id ) ) {
+  if( target->type != STUMPLESS_FILE_TARGET && unlikely( !target->id ) ) {
```

Comment:
After this NULL check is where I would add the check of ODELAY, and subsequent calls to `stumpless_target_is_open` and `stumpless_open_target`.